src/providers/tenderly-simulation-provider.ts

```diff
@@ -0,0 +1,106 @@
+import axios from 'axios'
+import { log } from '../util'
+import { APPROVE_TOKEN_FOR_TRANSFER, V3_ROUTER2_ADDRESS } from '../util/callData'
+
+export const TENDERLY_BATCH_SIMULATE_API = (
+  TENDERLY_BASE_URL: string,
+  TENDERLY_USER: string,
+  TENDERLY_PROJECT: string
+) => `${TENDERLY_BASE_URL}/api/v1/account/${TENDERLY_USER}/project/${TENDERLY_PROJECT}/simulate-batch`
+
+/**
+ * Provider for dry running transactions on tenderly.
+ *
+ * @export
+ * @interface ISimulator
+ */
+export interface ISimulator {
+  /**
+   * Returns the gas fee that was paid to land the transaction in the simulation.
+   * @returns number or Error
+   */
+  simulateTransaction: (
+    chainId: number,
+    hexData: string,
+    tokenInAddress: string,
+    fromAddress: string,
+    fallback?: number
+  ) => Promise<number|Error>
+}
+export class TenderlyProvider implements ISimulator {
+  TENDERLY_BASE_URL: string
+  TENDERLY_USER: string
+  TENDERLY_PROJECT: string
+  TENDERLY_ACCESS_KEY: string
```

Comment:
Variables within a class should me camelCase. Also set private visibility (default in ts is public)